lib/cc/cli/init.rb

```diff
@@ -74,27 +74,40 @@ def create_codeclimate_checks
 
       def create_default_engine_configs
         say "Generating default configuration for engines..."
-        available_engine_configs.each do |config_path|
-          file_name = File.basename(config_path)
-          if filesystem.exist?(file_name)
-            say "Skipping generating #{file_name} file (already exists)."
-          else
-            filesystem.write_path(file_name, File.read(config_path))
-            success "Config file #{file_name} successfully generated."
+        available_engine_configs.each do |(engine_name, config_paths)|
+          engine = engine_registry[engine_name]
+          config_mapping = Hash.new { |_, k| [k] }.merge(engine["config_files"] || {})
+
+          config_paths.each do |config_path|
+            generate_config(config_path, config_mapping[File.basename(config_path)])
           end
         end
       end
 
+      def generate_config(config_path, possible_names)
+        file_name = File.basename(config_path)
+        existing_files = possible_names.select do |f|
+          filesystem.exist? f
+        end
+        if existing_files.any?
+          say "Skipping generating #{file_name} file (already exists: #{existing_files.join(", ")})."
```

Comment:
I would reformat this a bit:

`Skipping generating #{file_name}, existing file(s) found: #{existing_files.join(", ")}).`